pkg/url/url.go

```diff
@@ -100,6 +104,74 @@ func GetIngress(kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInf
 	return iextensionsv1.Ingress{}, errors.New(fmt.Sprintf("the url %v does not exist", urlName))
 }
 
+// GetContainer returns Docker URL definition for given URL name
+func GetContainerURL(client *lclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, componentName string) (URL, error) {
+	localURLs := envSpecificInfo.GetURL()
+	containers, err := dockerutils.GetComponentContainers(*client, componentName)
+	if err != nil {
+		return URL{}, errors.Wrap(err, "unable to get component containers")
+	}
+	containerJSONMap := make(map[string]types.ContainerJSON)
+	for _, c := range containers {
+		containerJSON, err := client.Client.ContainerInspect(client.Context, c.ID)
+		if err != nil {
+			return URL{}, err
+		}
+		containerJSONMap[c.ID] = containerJSON
+		for internalPort, portbinding := range containerJSON.HostConfig.PortBindings {
+			remoteURLName := containerJSON.Config.Labels[internalPort.Port()]
+			if remoteURLName != urlName {
+				continue
+			}
+			externalport, err := strconv.Atoi(portbinding[0].HostPort)
+			if err != nil {
+				return URL{}, err
+			}
+			dockerURL := getMachineReadableFormatDocker(internalPort.Int(), externalport, portbinding[0].HostIP, remoteURLName)
+			for _, localurl := range localURLs {
+				if localurl.ExposedPort <= 0 {
+					continue
+				}
+				if localurl.Name == remoteURLName {
+					// URL is in both env file and docker host config
+					dockerURL.Status.State = StateTypePushed
+					return dockerURL, nil
+				}
+			}
+			dockerURL.Status.State = StateTypeLocallyDeleted
+			return dockerURL, nil
+		}
+	}
```

Comment:
Above code needs more comments. Lots of stuff happening here..

Reply:
updated